pkgs/development/compilers/bluespec-bsc/default.nix

```diff
@@ -0,0 +1,104 @@
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, autoconf
+, automake
+, bison
+, flex
+, fontconfig
+, ghc
+, glibc
+, gmp
+, gperf
+, haskell
+, libX11
+, libpoly
+, perl
+, pkgconfig
+, verilog
+, xorg
+, zlib
+}:
+
+let
+  gmp-static = gmp.override { withStatic = true; };
+  ghcWithPackages = haskell.packages.ghc844.ghc.withPackages (g: (with g; [old-time regex-compat syb]));
+in stdenv.mkDerivation rec {
+  pname = "bsc";
+  version = "unstable-2020.02.07";
+
+  src = fetchFromGitHub {
+    owner  = "B-Lang-org";
+    repo   = "bsc";
+    rev    = "553f21df9fa4703fe117720862997390024304d2";
+    sha256 = "1dqrkm8j2pki8hc4c5i4xl0jxba21y3wf589xrw8849yld6qgpki";
+    fetchSubmodules = true;
+  };
+
+  enableParallelBuilding = true;
```

Comment:
As I noted before, I've seen compilation failures with `-j` on high-core count machines (32 cores, for me), so we should just get rid of this, I feel. I'll try to debug it later or see if I can reproduce it...

Reply:
I had one unconclusive non-reproducible build failure:
```
[207 of 224] Compiling GenFuncWrap      ( GenFuncWrap.hs, /build/source/src/comp/../../build/GenFuncWrap.o )
[208 of 224] Compiling Depend           ( Depend.hs, /build/source/src/comp/../../build/Depend.o )
[209 of 224] Compiling AExpr2Yices      ( AExpr2Yices.hs, /build/source/src/comp/../../build/AExpr2Yices.o )
[210 of 224] Compiling SAT              ( SAT.hs, /build/source/src/comp/../../build/SAT.o )
[211 of 224] Compiling AProofs          ( AProofs.hs, /build/source/src/comp/../../build/AProofs.o )
[212 of 224] Compiling AOpt             ( AOpt.hs, /build/source/src/comp/../../build/AOpt.o )
[213 of 224] Compiling Synthesize       ( Synthesize.hs, /build/source/src/comp/../../build/Synthesize.o )
[214 of 224] Compiling SimPackageOpt    ( SimPackageOpt.hs, /build/source/src/comp/../../build/SimPackageOpt.o )
[215 of 224] Compiling LambdaCalcUtil   ( LambdaCalcUtil.hs, /build/source/src/comp/../../build/LambdaCalcUtil.o )
[216 of 224] Compiling SAL              ( SAL.hs, /build/source/src/comp/../../build/SAL.o )
[217 of 224] Compiling LambdaCalc       ( LambdaCalc.hs, /build/source/src/comp/../../build/LambdaCalc.o )
[218 of 224] Compiling ADumpSchedule    ( ADumpSchedule.hs, /build/source/src/comp/../../build/ADumpSchedule.o )
[219 of 224] Compiling DisjointTest     ( DisjointTest.hs, /build/source/src/comp/../../build/DisjointTest.o )
[220 of 224] Compiling ASchedule        ( ASchedule.hs, /build/source/src/comp/../../build/ASchedule.o )
[221 of 224] Compiling AState           ( AState.hs, /build/source/src/comp/../../build/AState.o )
[222 of 224] Compiling AAddSchedAssumps ( AAddSchedAssumps.hs, /build/source/src/comp/../../build/AAddSchedAssumps.o )
[223 of 224] Compiling ACleanup         ( ACleanup.hs, /build/source/src/comp/../../build/ACleanup.o )
[224 of 224] Compiling Main_bsc         ( bsc.hs, /build/source/src/comp/../../build/Main_bsc.o )
Linking bsc ...
Using flags:  @/tmp/ghc26512_0/ghc_818.rsp
Using flags:  @/tmp/ghc26512_0/ghc_821.rsp
bsc done Sat Feb 8 10:44:41 UTC 2020

make[2]: Leaving directory '/build/source/src/comp'
make[1]: *** [Makefile:13: install] Error 2
make[1]: Leaving directory '/build/source/src'
make: *** [Makefile:24: install] Error 2
builder for '/nix/store/krbznhgazzqjmjb57hgg8zbk3npdm3n3-bluespec-unstable-2020.02.07.drv' failed with exit code 2
```

That was on a 48 core machine, and the same derivation succeeded afterwards.

I'm not sure if that was caused due to parallel build, or just some flakiness in `bsc` itself. Could also be the actual error message is buried further up - I don't have the log anymore :-/

Did it look similar to you, or was it something different?

Reply:
Yeah, it was something like that, just an inscrutable failure. I haven't had any failures of this nature since I turned off parallel building, though. Superstitious, I know, but it's all I've got...

Reply:
It seems to happen quite seldomly, and I'm still not convinced it **doesn't** happen on a single-core build. I'm inclined to keep the parallel building in, as it dramatically increases compilation times (for the 95% of times it succeeds building), wdyt?

Reply:
Okay, I had another occurence, and decided to set `enableParallelBuilding = false;`, and opened https://github.com/B-Lang-org/bsc/issues/38.